pymc_experimental/distributions/multivariate.py

```diff
@@ -0,0 +1,246 @@
+    r2: tensor
+        expected R2 for the linear regression
+    phi: tensor
+        variance weights that sums up to 1
+    psi: tensor
+        probability of a coefficients to be positive
+    """
+    tau2 = r2 / (1 - r2)
+    explained_variance = phi * pt.expand_dims(tau2 * output_sigma**2, -1)
+    mu_param, std_param = _psivar2musigma(psi, explained_variance)
+    if not centered:
+        with pm.Model(name):
+            raw = pm.Normal("raw", dims=dims)
+        beta = pm.Deterministic(name, (raw * std_param + mu_param) / input_sigma, dims=dims)
+    else:
+        beta = pm.Normal(name, mu_param / input_sigma, std_param / input_sigma, dims=dims)
+    return beta
+
+
+def R2D2M2CP(
+    name,
+    output_sigma,
+    input_sigma,
+    *,
+    dims,
+    r2,
+    variables_importance=None,
+    variance_explained=None,
+    r2_std=None,
+    positive_probs=0.5,
+    positive_probs_std=None,
+    centered=False,
+):
+    """R2D2M2CP Prior.
+
+    Parameters
+    ----------
+    name : str
+        Name for the distribution
+    output_sigma : tensor
+        Output standard deviation
+    input_sigma : tensor
+        Input standard deviation
+    dims : Union[str, Sequence[str]]
+        Dims for the distribution
+    r2 : tensor
+        :math:`R^2` estimate
+    variables_importance : tensor, optional
+        Optional estimate for variables importance, positive, by default None
+    variance_explained : tensor, optional
+        Alternative estimate for variables importance which is point estimate of
+        variance explained, should sum up to one, by default None
+    r2_std : tensor, optional
+        Optional uncertainty over :math:`R^2`, by default None
+    positive_probs : tensor, optional
+        Optional probability of variables contribution to be positive, by default 0.5
+    positive_probs_std : tensor, optional
+        Optional uncertainty over effect direction probability, by default None
+    centered : bool, optional
+        Centered or Non-Centered parametrization of the distribution, by default Non-Centered. Advised to check both
+
+    Returns
+    -------
+    residual_sigma, coefficients
+        Output variance (sigma squared) is split in residual variance and explained variance.
+
+    Raises
+    ------
+    TypeError
+        If parametrization is wrong.
+
+    Notes
+    -----
+    The R2D2M2CP prior is a modification of R2D2M2 prior.
+
+    - ``(R2D2M2)``CP is taken from https://arxiv.org/abs/2208.07132
+    - R2D2M2``(CP)``, (Correlation Probability) is proposed and implemented by Max Kochurov (@ferrine)
+
+    Examples
+    --------
+    Here are arguments explained in a synthetic example
+
+    >>> import pymc_experimental as pmx
+    >>> import pymc as pm
+    >>> import numpy as np
+    >>> X = np.random.randn(10, 3)
+    >>> b = np.random.randn(3)
+    >>> y = X @ b + np.random.randn(10) * 0.04 + 5
+    >>> with pm.Model(coords=dict(variables=["a", "b", "c"])) as model:
```

Comment:
Maybe split the initialization from the first model, so users can easily run the second or third examples if they want?

Also I think we have a different docs formatting that allows to copy paste snippets directly

Reply:
https://github.com/pymc-devs/pymc-experimental/blob/e38da068d92e8464dc879790ecad69ebdf6d4398/pymc_experimental/marginal_model.py#L56